script.js

```diff
@@ -1,4 +1,145 @@
+  if (input == 1 && gameMode == "Player 2 rolls") {
+    diceOrderPlayer2 = Number("" + diceResultPlayer2[0] + diceResultPlayer2[1]);
+  } else if (input == 2 && gameMode == "Player 2 rolls") {
+    diceOrderPlayer2 = Number("" + diceResultPlayer2[1] + diceResultPlayer2[0]);
+  }
+  return diceOrderPlayer2;
+};
+
+//compare numbers, player with higher number wins
+var decideWinner = function (input, diceOrderPlayer1, diceOrderPlayer2) {
+  var diceOrderPlayer1 = decideDiceOrderPlayer1(input, diceResultPlayer1);
+  var diceOrderPlayer2 = decideDiceOrderPlayer2(input, diceResultPlayer2);
+  var winner = `It's a draw!`;
+  if (diceOrderPlayer1 > diceOrderPlayer2) {
+    winner = `Player 1 wins!`;
+  } else if (diceOrderPlayer2 > diceOrderPlayer1) {
+    winner = `Player 2 wins!`;
+  }
+  return winner;
+};
+
+// valid inputs are "", 1 or 2.
+var checkInputValid = function (input) {
+  inputValid = false;
+  if (
+    (input == "" && gameMode == `Player 1 rolls`) ||
+    input == 1 ||
+    input == 2
+  ) {
+    inputValid = true;
+  }
+  return inputValid;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var diceOrderPlayer1 = decideDiceOrderPlayer1(input, diceResultPlayer1);
+  var diceOrderPlayer2 = decideDiceOrderPlayer2(input, diceResultPlayer2);
+  // click submit and roll dice for Player 1
+  if (input == "" && gameMode == `Player 1 rolls`) {
+    diceResultPlayer1 = rollDiceResultPlayer1();
+    myOutputValue = `Player 1 rolled ${diceResultPlayer1}! <br><br> Press "1" to retain order or "2" to reverse order <br><br> E.g. You rolled 3 and 5. Press "1" for 35, and "2" for 53. `;
+    console.log("Player 1", diceResultPlayer1);
+  }
+  // Player 1 decides dice order
+  else if (
+    (input == 1 && gameMode == `Player 1 rolls`) ||
+    (input == 2 && gameMode == `Player 1 rolls`)
+  ) {
+    console.log("diceOrderPlayer 1", diceOrderPlayer1);
+    myOutputValue = `Player 1's hand: ${diceOrderPlayer1}<br><br> It is now Player 2's turn`;
+    gameMode = `Player 2 rolls`;
+    console.log(`gameMode`, gameMode);
+  }
+  // click submit and roll dice for Player 2
+  else if (input == "" && gameMode == `Player 2 rolls`) {
+    diceResultPlayer2 = rollDiceResultPlayer2();
+    console.log("Player 2", diceResultPlayer2);
+    myOutputValue = `Player 2 rolled ${diceResultPlayer2}! <br><br> Press "1" to retain order or "2" to reverse order <br><br> E.g. You rolled 3 and 5. Press "1" for 35, and "2" for 53. `;
+  }
+  // Player 2 decides dice order
+  else if (
+    (input == 1 && gameMode == `Player 2 rolls`) ||
+    (input == 2 && gameMode == `Player 2 rolls`)
+  ) {
+    console.log("diceOrderPlayer 2", diceOrderPlayer2);
+    myOutputValue = `Player 2's hand: ${diceOrderPlayer2}<br><br> Click "submit" to find out who wins!`;
+    gameMode = `decide winner`;
+    console.log(`gameMode`, gameMode);
+  }
+  // click submit and obtain winner
+  else if (input == "" && gameMode == `decide winner`) {
```

Comment:
gameMode == 'decide winner' would actually suffice in this case, because there are no other else statements which uses this game mode.